### PR TITLE
add ngc signing job for auto signing

### PR DESCRIPTION
**Pull request by shivakunv**
Hello!

Thanks for making this contribution! When contributing to this repository please keep in mind the following:
- [You should sign your work](https://github.com/NVIDIA/gpu-operator/blob/master/CONTRIBUTING.md).
- You should be making your contribution against the [gitlab.com repository](https://gitlab.com/nvidia/kubernetes/gpu-operator) as github.com is just a mirror.

### CODE REVIEW

**Comment by shivakunv** [comment]
@cdesiniotis @elezar  PTAL

File: .nvidia-ci.yml [hunk 1]
```diff
@@ -174,53 +174,53 @@ release:ngc-gpu-operator-validator:
...
+  image: ubuntu:latest
   stage: sign
+  rules:
+    - if: $CI_COMMIT_TAG
   variables:
-    IMAGE_NAME: "${OUT_IMAGE_NAME}"
-    VERSION: "${OUT_IMAGE_VERSION}"
+    NGC_CLI_API_KEY: "${NGC_REGISTRY_TOKEN}"
+    IMAGE_NAME: "${NGC_REGISTRY_IMAGE}"
+    IMAGE_TAG: "${CI_COMMIT_TAG}"
```
Comment (cdesiniotis) [inline .nvidia-ci.yml]:
```suggestion
    IMAGE_TAG: "${CI_COMMIT_TAG}-${DIST}"
```

File: .nvidia-ci.yml [hunk 2]
```diff
@@ -174,53 +174,53 @@ release:ngc-gpu-operator-validator:
...
-    - 'echo "Signing the image ${IMAGE_NAME}:${VERSION}"'
-    - 'echo "ngc registry image publish --source ${IMAGE_NAME}:${VERSION} ${IMAGE_NAME}:${VERSION} --public --discoverable --allow-guest --sign"'
+    - 'echo "Signing the image ${IMAGE_NAME}:${IMAGE_TAG}"'
+    - ngc-cli/ngc registry image publish --source ${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_NAME}:${IMAGE_TAG} --public --discoverable --allow-guest --sign --org nvidia
 
 sign:ngc-gpu-operator:
   extends:
     - .sign:ngc
   needs:
     - release:ngc-gpu-operator
```
Comment (cdesiniotis) [inline .nvidia-ci.yml:215]:
```suggestion
    - release:ngc-gpu-operator
  variables:
    IMAGE_TAG: "${CI_COMMIT_TAG}"
```

File: .nvidia-ci.yml [hunk 3]
```diff
@@ -174,53 +174,53 @@ release:ngc-gpu-operator-validator:
...
-  rules:
-    - if: $CI_COMMIT_TAG
-  variables:
-    OUT_IMAGE_VERSION: "${CI_COMMIT_TAG}"
-    OUT_IMAGE_NAME: "${NGC_REGISTRY_IMAGE}" # This needs to change for the gpu-operator and gpu-operator-validator
```
Comment (cdesiniotis) [inline .nvidia-ci.yml:225]:
```suggestion

sign:ngc-gpu-operator-ubi8:
  extends:
    - .sign:ngc
    - .dist-ubi8
  needs:
    - release:ngc-gpu-operator
    
```

File: .nvidia-ci.yml [hunk 4]
```diff
@@ -174,53 +174,53 @@ release:ngc-gpu-operator-validator:
...
   extends:
     - .sign:ngc
   needs:
     - release:ngc-gpu-operator-validator
-  rules:
-    - if: $CI_COMMIT_TAG
   variables:
-    OUT_IMAGE_VERSION: "${CI_COMMIT_TAG}"
-    OUT_IMAGE_NAME: "${NGC_PROD_VALIDATOR_IMAGE}"
+    IMAGE_NAME: "${NGC_PROD_VALIDATOR_IMAGE}"
```
Comment (cdesiniotis) [inline .nvidia-ci.yml]:
```suggestion
    IMAGE_NAME: "${NGC_PROD_VALIDATOR_IMAGE}" 
    IMAGE_TAG: "${CI_COMMIT_TAG}"
```

File: .nvidia-ci.yml [hunk 5]
```diff
@@ -174,53 +174,53 @@ release:ngc-gpu-operator-validator:
...
```
Comment (cdesiniotis) [inline .nvidia-ci.yml:243]:
```suggestion

sign:ngc-gpu-operator-validator-ubi8:
  extends:
    - .sign:ngc
    - .dist-ubi8
  needs:
    - release:ngc-gpu-operator-validator
  rules:
    - if: $CI_COMMIT_TAG
  variables:
    IMAGE_NAME: "${NGC_PROD_VALIDATOR_IMAGE}" 
    
```

**Review by cdesiniotis — COMMENTED** [review]
@shivakunv For both the gpu-operator and gpu-operator-validator images, we push a short tag and a tag with the `ubi8` suffix. I left some suggestions so that we sign both tags now. These changes align with the changes you made in the other projects.

File: .nvidia-ci.yml (hunk 2 above)
Comment (cdesiniotis) [reply .nvidia-ci.yml:215]: This configures the short tag.

File: .nvidia-ci.yml (hunk 3 above)
Comment (cdesiniotis) [reply .nvidia-ci.yml:225]: This is a separate sign job to sign the `ubi8` tag.

File: .nvidia-ci.yml (hunk 4 above)
Comment (cdesiniotis) [reply .nvidia-ci.yml]: This configures the short tag.

File: .nvidia-ci.yml (hunk 5 above)
Comment (cdesiniotis) [reply .nvidia-ci.yml:243]: This is a separate sign job to sign the `ubi8` tag.

File: .nvidia-ci.yml (hunk 1 above)
Comment (shivakunv) [reply .nvidia-ci.yml]: taken

File: .nvidia-ci.yml (hunk 2 above)
Comment (shivakunv) [reply .nvidia-ci.yml:215]: taken

File: .nvidia-ci.yml (hunk 3 above)
Comment (shivakunv) [reply .nvidia-ci.yml:225]: taken

File: .nvidia-ci.yml (hunk 4 above)
Comment (shivakunv) [reply .nvidia-ci.yml]: taken

File: .nvidia-ci.yml (hunk 5 above)
Comment (shivakunv) [reply .nvidia-ci.yml:243]: taken. 
removed rules 
  rules:
    - if: $CI_COMMIT_TAG

as it is already imported in .sign:ngc

**Comment by shivakunv** [comment]
@cdesiniotis  resolved all comments PTAL

**Review by cdesiniotis — APPROVED** [review]

**Pull request merged**